src/Plugin/views/field/EntityBundle.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\commerce\Plugin\views\field\EntityBundle.
+ */
+
+namespace Drupal\commerce\Plugin\views\field;
+
+use Drupal\views\Plugin\views\field\FieldPluginBase;
+use Drupal\views\ResultRow;
+
+/**
+ * Field handler to display the entity bundle. It displays nothing if there's
+ * only one bundle. This is useful if used together with the 'Hide empty
+ * column' setting of the Table format plugin.
+ *
+ * @ingroup views_field_handlers
+ *
+ * @ViewsField("entity_bundle")
+ */
+class EntityBundle extends FieldPluginBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function query() {
+    // Do not change the query.
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function render(ResultRow $row) {
+    // Get entity.
+    $entity_type = $this->getEntityType();
+    // Get the bundles of this entity.
+    $bundles = \Drupal::service('entity_type.bundle.info')->getBundleInfo($entity_type);
+
+    // If there's only one bundle, don't dislay it.
```

Comment:
There should be a setting (enabled by default) in the views handler for this behavior, so that it can be turned off.